src/librustc/dep_graph/dep_tracking_map.rs

```diff
@@ -134,7 +175,7 @@ impl<M: DepTrackingMapConfig> MemoizationMap for RefCell<DepTrackingMap<M>> {
     /// accesses the body of the item `item`, so we register a read
     /// from `Hir(item_def_id)`.
     fn memoize<OP>(&self, key: M::Key, op: OP) -> M::Value
-        where OP: FnOnce() -> M::Value
+        where OP: FnOnce() -> Option<M::Value>
```

Comment:
It's a little hard to see because of how the diff reads on Github, but [this call to memoize](https://github.com/rust-lang/rust/pull/40362/files#diff-c888afeccd78f53e5ea3eb04089e46ddL2056) has a closure that returns `None`.